src/main/java/yevhen/bookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,84 @@
+package yevhen.bookstore.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import yevhen.bookstore.dto.book.BookDtoWithoutCategoryIds;
+import yevhen.bookstore.dto.category.CategoryDto;
+import yevhen.bookstore.dto.category.CreateCategoryRequestDto;
+import yevhen.bookstore.service.book.BookService;
+import yevhen.bookstore.service.category.CategoryService;
+
+@Tag(name = "Book categories managing", description = "Endpoints for managing book categories")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    @Operation(summary = "Create a new book category", description = "Create a new book category")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    public CategoryDto createCategory(@RequestBody @Valid CreateCategoryRequestDto requestDto) {
+        return categoryService.save(requestDto);
+    }
+
+    @GetMapping
+    @Operation(summary = "Get all book categories",
+            description = "Get a list of all available book categories")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_USER')")
```

Comment:
1)You can use it without the prefix ROLE_
2) I guess that each admin also has the role User  because  you add it in the method `register()` (check other methods )

```suggestion
    @PreAuthorize("hasAnyRole("('ROLE_USER')")
```